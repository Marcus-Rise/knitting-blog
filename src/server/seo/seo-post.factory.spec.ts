import { SeoPostFactory } from "./seo-post.factory";
import { mock } from "jest-mock-extended";
import type { IPostWithoutContent } from "../../common/post";

describe("SeoPostFactory", () => {
  describe("toSitemapRaw", () => {
    it("should escape symbols", () => {
      const { slug, date } = mock<IPostWithoutContent>({
        slug: "коричневый-кардиган",
        date: new Date(2020, 12, 12).toJSON(),
      });
      const item = SeoPostFactory.toSitemapRaw({ slug, date, hostName: "https://test.com" });

      expect(item).not.toContain(slug);
      expect(item).toMatchSnapshot();
    });
  });
});
