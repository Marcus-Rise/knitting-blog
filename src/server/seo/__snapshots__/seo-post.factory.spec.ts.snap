// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeoPostFactory toSitemapRaw should escape symbols 1`] = `
"<url>
      <loc>https://test.com/%D0%BA%D0%BE%D1%80%D0%B8%D1%87%D0%BD%D0%B5%D0%B2%D1%8B%D0%B9-%D0%BA%D0%B0%D1%80%D0%B4%D0%B8%D0%B3%D0%B0%D0%BD/</loc>
      <lastmod>2021-01-11T20:00:00.000Z</lastmod>
      </url>"
`;
